src/core.c/Distribution/Resource.pm6

```diff
@@ -4,81 +4,31 @@ class Distribution::Resource {
     has $.dist-id;
     has $.key;
 
-    method IO() {
+    # delegate appropriate IO::Path methods to the resource IO::Path object
+    has IO::Path $.IO is built(False) handles <
```

Comment:
This is breaking installation via Staging repository. IO() **must** be determined at runtime. The result **must not** make it into a precomp file.
Cf. commit 67906e48dca69ff9af9dbb4d01dd1ff3a1e759b9 and d4d6a9976403c7a8339aaf44d3fb9e9656c28232

Reply:
FWIW, then we must add some tests somewhere, because this was spectest clean.